src/nvim/terminal.c

```diff
@@ -250,12 +253,16 @@ Terminal *terminal_open(TerminalOptions opts)
   rv->sb_size = MIN(rv->sb_size, 100000);
   rv->sb_buffer = xmalloc(sizeof(ScrollbackLine *) * rv->sb_size);
 
+  if (!true_color) {
+      return rv;
+  }
+
   // Configure the color palette. Try to get the color from:
   //
   // - b:terminal_color_{NUM}
   // - g:terminal_color_{NUM}
   // - the VTerm instance
```

Comment:
I did this because it you are not in true color mode you already know which color id to output (they are stored and inited into color_indexes).

And because the g:terminal_color colors wouldn't be used if we are not in true color mode, I simply skipped over them.

I also now use the native libvterm way to set the default colors. This is a lot better and fixes the problem where a true color code would be incorrectly converted to an indexed color. 

However because I use the native way to set colors, it will mess up the color_indexes map. This will only become a problem in none true color mode though, so that is why I added the return.

I'm not sure how I would write all that into a comment though... Is "Only set custom terminal colors if in true color mode" enough?